client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+import { DraftailEditor, ToolbarButton, createEditorStateFromRaw, serialiseEditorStateToRaw } from 'draftail';
+import { EditorState, Modifier, RichUtils, SelectionState } from 'draft-js';
+import { filterInlineStyles } from "draftjs-filters";
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+
+import { STRINGS } from '../../../config/wagtailConfig';
+import Icon from '../../Icon/Icon';
+
+const COMMENT_STYLE_IDENTIFIER = 'COMMENT-'
+
+function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]); // Only re-run if value changes
+  return ref.current;
+}
+
+class DraftailInlineAnnotation {
+    constructor(field) {
+      this.field = field;
+      this.inlineRefs = new Set()
+    }
+    addRef(ref) {
+      this.inlineRefs.add(ref);
+    }
+    removeRef(ref) {
+      this.inlineRefs.delete(ref);
+    }
+    getDesiredPosition() {
+      const nodeTops = Array.from(this.inlineRefs).map(ref => ref.current).filter(node => node).map(node => node.getBoundingClientRect().top)
+      if (nodeTops.length > 0) {
+        return nodeTops.reduce((a,b) => a + b, 0)/nodeTops.length + document.documentElement.scrollTop
+      }
+      const fieldNode = this.field
+      if (fieldNode) {
+        return fieldNode.getBoundingClientRect().top + document.documentElement.scrollTop
+      }
+      return 0
+    }
+}
+
+function getFullSelectionState(contentState) {
+  // Get a selection state corresponding to the full contentState
+  const lastBlock = contentState.getLastBlock();
+  let fullSelectionState = SelectionState.createEmpty();
+  fullSelectionState = fullSelectionState.set('anchorKey', contentState.getFirstBlock().getKey());
+  fullSelectionState = fullSelectionState.set('focusKey', lastBlock.getKey());
+  fullSelectionState = fullSelectionState.set('anchorOffset', 0);
+  fullSelectionState = fullSelectionState.set('focusOffset', lastBlock.getLength());
+  return fullSelectionState
+}
+
+function getCommentControl(commentApp, contentPath, fieldNode) {
+  return ({getEditorState, onChange}) => {
+    return <ToolbarButton
+      name='comment'
+      active={false}
+      title={STRINGS.ADD_A_COMMENT}
+      icon={<Icon name="comment"/>}
+      onClick={() => {
+        const annotation = new DraftailInlineAnnotation(fieldNode)
+        const commentId = commentApp.makeComment(annotation, contentPath);
+        onChange(RichUtils.toggleInlineStyle(getEditorState(), COMMENT_STYLE_IDENTIFIER + commentId))
```

Comment:
Only use `+` for math.

```suggestion
        onChange(RichUtils.toggleInlineStyle(getEditorState(),`${COMMENT_STYLE_IDENTIFIER}${commentId}`))
```

Reply:
Thanks, will fix!